cmd/opampsupervisor/e2e_test.go

```diff
@@ -699,6 +731,31 @@ func TestSupervisorOpAMPConnectionSettings(t *testing.T) {
 	}, 10*time.Second, 500*time.Millisecond, "Collector did not connect to new OpAMP server")
 }
 
+func TestSupervisorCanStartWithoutServerThenConnectLater(t *testing.T) {
+	var connectedToServer atomic.Bool
+	connectedToServer.Store(false)
+	initialServer := newUnstartedOpAMPServer(
+		t,
+		defaultConnectingHandler,
+		server.ConnectionCallbacksStruct{
+			OnConnectedFunc: func(_ context.Context, _ types.Connection) {
+				connectedToServer.Store(true)
+			},
+		})
+	defer initialServer.shutdown()
+
+	s := newSupervisor(t, "accepts_conn", map[string]string{"url": initialServer.addr})
+	defer s.Shutdown()
+	time.Sleep(2 * time.Second) // We wait until the supervisor gives up on connecting
+	require.False(t, connectedToServer.Load(), "Collector connected to server before server was started")
+	require.True(t, s.GetAgentDescription() != nil, "Agent description was not received, so agent may not have started.")
```

Comment:
Instead of exposing a public method, could we check that the Collector reports as healthy? You can see an example in `TestSupervisorRestartCommand`. It's a bit more code, but will allow us to forgo the additional method and I think is a bit more of a direct test that the Collector is live.

Reply:
the code in that test relies on our own opamp server receiving a connection from the agent. in our test, we don't have a server initially, right? The only server is implemented inside the supervisor. That's why I added the public method - to expose state from the internal server implemented by the supervisor.

I agree that a new public method feels like too much - but I don't know another reasonable way to get that internal state. Thoughts?
on the other hand, the Supervisor component is not meant to be a library, so we could have a lower bar for allowing changes to its public api.